### PR TITLE
Incorporate Bootswatch themes

### PR DESCRIPTION
**Pull request by jason-fox**
Builds on https://github.com/infotexture/dita-bootstrap/pull/91

Additional `--bootstrap.theme` parameter to download [Bootswatch](https://bootswatch.com/) themes

#### --bootstrap.theme=morph 

Screenshot 2023-06-24 at 10 17 57

####  --bootstrap.theme=sketchy

Screenshot 2023-06-24 at 10 24 54

####  --bootstrap.theme=cerulean

Screenshot 2023-06-24 at 10 19 04

####  --bootstrap.theme=lux

Screenshot 2023-06-24 at 10 20 13

` --bootstrap.theme=nome` also removes the default CSS when customising  using `--args.css` and not using the default Bootstrap CSS.

### CODE REVIEW

**Review by infotexture — COMMENTED** [review]
🙏 Thanks for this initiative. I've often thought about making it easier to customize things here via Sass, but wasn't sure whether the additional complexity was worth it, or if users would prefer to edit a single CSS file.

In the early versions of the plug-in, the `custom.css` file was just a few lines, which made it very straightforward to follow and update.

The addition of Themestr.app styles in 8bac0bb6988633b049af4a273b48384d95e4a708 helped to demonstrate the range of things that could be customized via this approach. The longer stylesheet seemed a fair price to pay.

8076cd8 added `css/style.css` and began requiring `fox.jason.extend.css`, which enabled new use cases, but also made it a bit trickier for users to figure out where to customize styling.

Personally, I prefer the Sass-based workflows, and these work well with Bootstrap's setup, but they do require the extra compilation step. 

I think that's OK given the extra capabilities they provide, but I'm beginning to wonder if we should keep adding things, or take this as an opportunity to consolidate in a few places.

## Goals

- For those who are fine with vanilla Bootstrap output, I'd like this to work out of the box without further customization or build steps.
- For many, tweaking a few CSS variables may be enough, so we should make that easy.
- For others who are comfortable with Sass & NPM builds, exposing more complex capabilities for overrides would probably be very welcome.

## Questions

1. If we add this, what could we change elsewhere?
2. Shall we drop the Themestr.app example in `custom.css` in favor of Bootswatch Sass?
3. If we have both, which wins?
4. Should the utility stylesheets in `css/` be Sass modules?
5. Do we still need `fox.jason.extend.css`, or could the Sass workflow replace that?

I realize these are things that may be better handled in separate PRs, but this seemed like as good a place as any to start the discussion.

File: sass/package.json
[the remark this answers is not in the document]
Comment (jason-fox) [reply sass/package.json:1]: Yes, this is a separate SASS Compilation utility. The bootswatch and sass modules are needed to compile a custom theme via Node.js

**Comment by jason-fox** [comment]
> ## Goals
> * For those who are fine with vanilla Bootstrap output, I'd like this to work out of the box without further customization or build steps.

If you leave out `--bootstrap.theme` and `--args.css` the result is Vanilla Bootstrap. The only additional injected CSS is the grid for mobile responsive HTML.

For a miminal but pleasing change, if you add `--bootstrap.theme`  you just get one of the standard Bootswatch themes, no fuss. The only additional injected CSS is the grid for mobile responsive HTML.

> * For many, tweaking a few CSS variables may be enough, so we should make that easy.

`--args.css` can be used to add some simple custom changes. e.g.  Vanilla Bootstrap + grid + minor changes

> * For others who are comfortable with Sass & NPM builds, exposing more complex capabilities for overrides would probably be very welcome.

SASS folder is for complex changes. Assumes you have Node.js available:

1. Edit `overrides.scss`
2. Run `npm install` 
3. Run `npm run sass` 
4.  Build DITA with. `--bootstrap.theme=none --args.css=theme ` 

Note that for simplicity, `overrides.scss` currently just offers the color palette SASS variables. If you really want to go advanced you could override any of the Bootstrap SASS variables.

**Comment by infotexture** [comment]
OK, thanks for the explanations above. The out-of-the-box Bootswatch themes work great. 🙂 

> https://themestr.app/ is not available online for me. I can't tell if they have updated to 5.3.0. If not it could be dropped.

Nor for me. 

I'd prefer to remove this from `custom.css` and use the Bootswatch themes and Sass-based customization options instead.

> 4. Build DITA with. `--bootstrap.theme=none --args.css=theme`

I think we need to clarify how to use the result of the `npm run sass` script.

I tested this by changing the primary color value in `sass/override.scss` and running the script to compile `sass/theme.css`, but `--bootstrap.theme=none --args.css=sass/theme.css` did't include the modified styles in the rendered output.

File: README.md [hunk 1]
```diff
@@ -124,16 +124,45 @@ dita --input=path/to/your.ditamap \
...
+from SASS based on the default Bootstrap theme (or any of the Bootswatch themes) can be
+found [here](#21). To run the tool, follow the instructions found within
+`sass/override.scss` file and run:
+
+```console
+npm install
+npm run sass
+```
+
+A new generated CSS theme will be found at `sass/theme.css`.
```
Comment (infotexture) [inline README.md:151]: We should probably explain here how to pass this generated file to the `dita` command.

File: README.md [hunk 2]
```diff
@@ -124,16 +124,45 @@ dita --input=path/to/your.ditamap \
...
+
+Additional Bootstrap themes can also be generated via [Themestr.app][9]. The plug-in
+includes a sample placeholder for [custom CSS][10] styles. You can edit the
+`css/custom.css` file to replace the theme, or add style rules of your own.
 
-To override the default theme, pass a custom CSS file to the `dita` command via the `--args.css` parameter:
+To override the default theme, pass a custom CSS file to the `dita` command via the
+`--args.css` parameter, should the default bootstrap CSS is no be longer required,
```
Comment (infotexture) [inline README.md:158]:
```suggestion
`--args.css` parameter, should the default Bootstrap CSS no longer be required,
```

File: sample/index.dita [hunk 3]
```diff
@@ -215,26 +215,64 @@
...
+            href="https://github.com/infotexture/dita-bootstrap/blob/master/css/custom.css"
+            format="html"
+            scope="external"
+          >custom CSS</xref> styles. You can edit the <filepath>css/custom.css</filepath> file to
+          replace the theme, or add style rules of your own.</p>
+      </note>
       <p>To override the default Bootstrap theme, pass a custom CSS file like this to the <cmdname>dita</cmdname>
-        command via the <parmname>--args.css</parmname> parameter:</p>
+        command via the <parmname>--args.css</parmname> parameter,
+        should the default bootstrap CSS is no be longer required, set
```
Comment (infotexture) [inline sample/index.dita:269]:
```suggestion
        should the default Bootstrap CSS no longer be required, set
```

File: sass/index.html [hunk 4]
```diff
@@ -0,0 +1,104 @@
+<!-- index.html -->
```
Comment (infotexture) [inline sass/index.html:1]: The docs don't seem to mention this new file.

I assume it's intended for testing the generated theme, but we should probably explain that somewhere.

File: README.md (hunk 1 above)
Comment (jason-fox) [reply README.md]: I've decided this would be better to be explained as a separate plugin

File: README.md (hunk 2 above)
Comment (jason-fox) [reply README.md]: Outdated - text has been updated

File: sample/index.dita (hunk 3 above)
Comment (jason-fox) [reply sample/index.dita]: See above

File: sass/index.html (hunk 4 above)
Comment (jason-fox) [reply sass/index.html]: This will be explained in a separate plugin

**Comment by jason-fox** [comment]
I've reduced this PR to add an extension-point.  SASS generation has been moved to a separate dependent [dita-bootstrap.sass](https://github.com/jason-fox/dita-bootstrap.sass)  plugin.  If you agree I will move it over here to keep the plugins together.

**Review by infotexture — APPROVED** [review]
👍 Makes sense to move Sass customization to a new plug-in.

I think that feature will be very useful, but not everyone that uses this will need it, so probably best to spin it off like the Lunr search support.

**Comment by infotexture** [comment]
@jason-fox The repo transfer invitation had expired by the time I got the message. ⏳ 

Happy to accept if you can re-initiate the transfer process.

**Comment by jason-fox** [comment]
@infotexture Done

**Pull request merged**